middleware/gzip/filter.go

```diff
@@ -72,7 +84,7 @@ func DefaultMIMEFilter() MIMEFilter {
 // matches any of the registered ones. It returns true if
 // found and false otherwise.
 func (m MIMEFilter) ShouldCompress(r *http.Request) bool {
-	return m.Types.Contains(r.Header.Get("Content-Type"))
+	return m.Types.Contains(r.Header.Get("Accept"))
```

Comment:
I agree with that - and good catch, you're right. I was pretty tired too. :smile: Doesn't using `mime.TypeByExtension()` effectively do the same thing as the extension filter?